openassessment/xblock/test/test_submission.py

```diff
@@ -1328,6 +1328,57 @@ def test_change_team_with_submission(self, xblock):
         self.assertEqual(path, 'openassessmentblock/response/oa_response_submitted.html')
         self.assertEqual(context['student_submission'], create_submission_dict(individual_submission, xblock.prompts))
 
+    @scenario('data/team_submission.xml', user_id="Red Five")
+    def test_team_has_already_submitted(self, xblock):
+        """
+        Test that if a user is on a team hat has already submitted, but they themself have not submitted,
+        the correct page is rendered.
+        """
+        self.setup_mock_team(xblock)
+
+        # pylint: disable=protected-access
+        xblock.runtime._services['user'] = NullUserService()
+        xblock.runtime._services['user_state'] = UserStateService()
+        xblock.runtime._services['teams'] = MockTeamsService(True)
+
+        # Create a submission for the team, but without r5.
+        arbitrary_user = get_user_model().objects.create_user(username='someuser', password='asdfasdfasf')
+        self._create_team_submission_and_workflow(
+            'test_course',
+            xblock.scope_ids.usage_id,
+            MOCK_TEAM_ID,
+            arbitrary_user.id,
+            ['rl', 'r2'],
+            {'text': 'This is the answer'},
+        )
+
+        # Assert that we return the 'team already submitted' path
+        self._assert_path_and_context(
```

Comment:
Random thought: would love to see these `asserts` moved into a helper class and, more specifically here, have one for just `assert_path` because I don't know that we care about much of the context here.